Code/ForceField/ANI/ANIForceField_catch.cpp

```diff
@@ -0,0 +1,435 @@
+      auto atom = conf.getAtomPos(i);
+      ps.push_back(&atom);
+      pos[3 * i] = atom.x;
+      pos[3 * i + 1] = atom.y;
+      pos[3 * i + 2] = atom.z;
+
+      ForceFields::ANI::ANIAtomContrib *ac;
+      ac = new ForceFields::ANI::ANIAtomContrib(
+          &field, speciesVec(i), i, speciesVec, numAtoms, 4, 8, "ANI-1x");
+      field.contribs().push_back(ForceFields::ContribPtr(ac));
+    }
+    field.initialize();
+    CHECK(std::fabs(field.calcEnergy(pos) - (-40.0517)) < 0.05);
+  }
+  SECTION("NH3") {
+    std::string pathName = getenv("RDBASE");
+
+    std::string molFile =
+        pathName + "/Code/GraphMol/Descriptors/test_data/NH3.mol";
+
+    RDKit::ROMOL_SPTR mol(RDKit::MolFileToMol(molFile, true, false));
+    int confId = -1;
+    auto conf = mol->getConformer(confId);
+    auto numAtoms = mol->getNumAtoms();
+    auto aev = RDKit::Descriptors::ANI::AtomicEnvironmentVector(*mol, confId);
+    auto speciesVec = RDKit::Descriptors::ANI::GenerateSpeciesVector(*mol);
+    ForceFields::ForceField field;
+    auto &ps = field.positions();
+    double *pos;
+    pos = new double[numAtoms * 3];
+    for (unsigned int i = 0; i < mol->getNumAtoms(); i++) {
+      auto atom = conf.getAtomPos(i);
+      ps.push_back(&atom);
+      pos[3 * i] = atom.x;
+      pos[3 * i + 1] = atom.y;
+      pos[3 * i + 2] = atom.z;
+
+      ForceFields::ANI::ANIAtomContrib *ac;
+      ac = new ForceFields::ANI::ANIAtomContrib(
+          &field, speciesVec(i), i, speciesVec, numAtoms, 4, 8, "ANI-1x");
+      field.contribs().push_back(ForceFields::ContribPtr(ac));
+    }
+    field.initialize();
+    CHECK(std::fabs(field.calcEnergy(pos) - (-56.1592)) < 0.05);
+  }
+  SECTION("Ethanol") {
+    std::string pathName = getenv("RDBASE");
+
+    std::string molFile =
+        pathName + "/Code/GraphMol/Descriptors/test_data/ethanol.sdf";
+
+    RDKit::ROMOL_SPTR mol(RDKit::MolFileToMol(molFile, true, false));
+    int confId = -1;
+    auto conf = mol->getConformer(confId);
+    auto numAtoms = mol->getNumAtoms();
+    auto aev = RDKit::Descriptors::ANI::AtomicEnvironmentVector(*mol, confId);
+    auto speciesVec = RDKit::Descriptors::ANI::GenerateSpeciesVector(*mol);
+    ForceFields::ForceField field;
+    auto &ps = field.positions();
+    double *pos;
+    pos = new double[numAtoms * 3];
+    for (unsigned int i = 0; i < mol->getNumAtoms(); i++) {
+      auto atom = conf.getAtomPos(i);
+      ps.push_back(&atom);
+      pos[3 * i] = atom.x;
+      pos[3 * i + 1] = atom.y;
+      pos[3 * i + 2] = atom.z;
+
+      ForceFields::ANI::ANIAtomContrib *ac;
+      ac = new ForceFields::ANI::ANIAtomContrib(
+          &field, speciesVec(i), i, speciesVec, numAtoms, 4, 8, "ANI-1x");
+      field.contribs().push_back(ForceFields::ContribPtr(ac));
+    }
+    field.initialize();
+    CHECK(std::fabs(field.calcEnergy(pos) - (-154.9892)) < 0.05);
+  }
+}
+TEST_CASE("QM7 Test Cases") {
+  SECTION("3C 1N and 5H Molecule") {
```

Comment:
In the QM7 dataset file I got(http://quantum-machine.org/datasets/), there were only NumPy arrays of the atomic coordinates and atom types but no information about bonds in the molecule and please correct me if I'm wrong, mol blocks require bond information as well.

Reply:
Ah, right, I forget the foibles of that dataset.
It would have required looking at the structures and figuring out where/what the bond orders are manually, which is almost certainly more trouble than it's worth.